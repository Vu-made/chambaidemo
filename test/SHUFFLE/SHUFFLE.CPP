#include <bits/stdc++.h>
#define N 100005
using namespace std;
int Next[N], Pre[N];
int n,x;
int main()
{
    freopen("SHUFFLE.INP","r",stdin);
    freopen("SHUFFLE.OUT","w",stdout);
    cin >> n >> x;
    Next[0] = 1; Pre[n+1] = n;
    for (int i=1;i<=n; i++)
    {
        Next[i] = i+1;
        Pre[i] = i-1;
    }
    for (int i,j,k=1;k <= x;k++)
    {
        cin >> i >> j;
        Next[Pre[i]] = Next[i];
        Pre[Next[i]] = Pre[i];
        Next[i] = j;
        Pre[i] = Pre[j];
        Next[Pre[j]] = i;
        Pre[j] = i;
    }
    for (int p=Next[0]; p<n+1; p=Next[p])
        cout << p << " ";
    return 0;
}
