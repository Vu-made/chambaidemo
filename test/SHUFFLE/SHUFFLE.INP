9 3
8 2
4 7
1 9
