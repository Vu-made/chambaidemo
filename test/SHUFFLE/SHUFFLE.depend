# depslib dependency file v1.0
1602671316 source:c:\users\admin\desktop\bgdv\bt\thuat_toan\array1d_3\shuffle\shuffle.cpp
	<bits/stdc++.h>

