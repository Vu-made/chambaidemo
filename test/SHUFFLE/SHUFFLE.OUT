8 2 3 5 6 4 7 1 9
8 2 3 5 6 4 7 1 9
